cinder/volume/drivers/vmware/volumeops.py

```diff
@@ -984,6 +984,33 @@ def get_summary(self, datastore):
                                         self._session.vim, datastore,
                                         'summary')
 
+    def get_datastore_custom_attributes(self, datastore):
+        retrieve_fields = self._session.invoke_api(vim_util,
+                                                   'get_object_property',
+                                                   self._session.vim,
+                                                   datastore,
+                                                   'availableField')
+        if retrieve_fields:
+            custom_fields = {}
+            for field in retrieve_fields:
+                for v in field[1]:
```

Comment:
Where is this `[1]` coming from? Could we document that, if it's really necessary?

Reply:
this has been removed.   I'm getting all the info from 1 api call now.